atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientInstrumentation.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.cassandra;
+
+public interface CassandraClientInstrumentation extends AutoCloseable {
+    void recordCellsWritten(String tableRef, long cellsWritten);
+
+    @Override
+    default void close() {}
```

Comment:
Only one implementation will need to do something here (close an executor). Also, I override the declaration to have it so no checked exception is assumed to be thrown by the interface method.